### ISSUE TITLE
Job Upload Date and Time Incorrect in QIDI Slicer

### ISSUE THREAD

**Issue by Cathobo** [QIDITECH/QIDISlicer#19]
I have had this problem with all versions of QIDI Slicer. In the Jobs Section under Modified the date is show as Oct 07 2023 and not Dec 31 2023. 
I contacted support and was given a terminal to correct the issue. It did but after turning my machine off and the rebooting the problem reappeared. 
The time zone on my pc is correct and the date and time format is correct in settings.

**Comment by Phil1988** [comment]
Is this coming from the slicer or from the printer?
Please check the .gcode file if the date (Oct 07) is anywhere in it.

If NOT, then its coming from the printer.
Please update the time on your maschine as I mentioned here:
https://github.com/QIDITECH/QIDI_MAX3/issues/15#issuecomment-1863875835

**Comment by Cathobo** [comment]
Thanks PhilThe date in the g code is correct. Support gave me a terminal called MobaXterm and a password and a file to flash the printer. Unfortunately it did not work. The printer is running at present . I will try your method .Neil
    On Sunday, 31 December 2023 at 11:58:54 GMT, Phil1988 ***@***.***> wrote:  
 
 


Is this coming from the slicer or from the printer?
Please check the .gcode file if the date (Oct 07) is anywhere in it.

If NOT, then its coming from the printer.
Please update the time on your maschine as I mentioned here:
QIDITECH/QIDI_MAX3#15 (comment)

**Comment by Cathobo** [comment]
I managed to connect to QIDI via my windows command prompt and changed the date using Sudo date, time and year, worked ok. I uploaded several gcode file and the time was showing correct. I then turned the QIDI off and waited a few minutes then tried again, all worked ok. I tried logging into my machine with root but was denied. I had to use makerbase.I then tried the sudo chmod command but came back no directory exists.I noticed when mks opened in my terminal the date last opened was Oct 7 2023, I will see what happens tomorrow. I think it will revert back to Oct 7.
    On Sunday, 31 December 2023 at 13:50:11 GMT, Neil Hobbs ***@***.***> wrote:  
 
  Thanks PhilThe date in the g code is correct. Support gave me a terminal called MobaXterm and a password and a file to flash the printer. Unfortunately it did not work. The printer is running at present . I will try your method .Neil
    On Sunday, 31 December 2023 at 11:58:54 GMT, Phil1988 ***@***.***> wrote:  
 
 


Is this coming from the slicer or from the printer?
Please check the .gcode file if the date (Oct 07) is anywhere in it.

If NOT, then its coming from the printer.
Please update the time on your maschine as I mentioned here:
QIDITECH/QIDI_MAX3#15 (comment)

**Comment by Phil1988** [comment]
@Cathobo : just use the commands I directly linked. The `sudo date` will only temporaly work - untill you shut the printer off.

Where exactly doesnt it find the directory?
You may not have ntpd installed.

First: Use putty. (no cmd, no other software)
login as "mks" and use the password

use the 4 lines code I provided (one at a time).
If it fails at `sudo chmod -x /usr/sbin/ntpd` then ntp is not installed.

Then you have to insall it first. It will only work if your current time beeing correct, because apt-get needs correct time.
So set the date temporaly with `sudo date` and then do
`sudo apt-get install ntp`

After this is done, proceed with the given command lines I gave you :)

**Comment by Cathobo** [comment]
Hi PhilYou are right the date and time was only temp.I have got ntp version 1:4.2.8p12+dfsg-4 installed.I used the sudo date again but when I used sudo chmod -x I get no such file or directory.Using sudo help I find -c -f -v and -R but no -x.Neil

Sent from Yahoo Mail on Android 
 
  On Mon, 1 Jan 2024 at 10:26, ***@***.***> wrote:   


@Cathobo : just use the commands I directly linked. The sudo date will only temporaly work - untill you shut the printer off.

Where exactly doesnt it find the directory?
You may not have ntpd installed.

First: Use putty. (no cmd, no other software)
login as "mks" and use the password

use the 4 lines code I provided (one at a time).
If it fails at sudo chmod -x /usr/sbin/ntpd then ntp is not installed.

Then you have to insall it first. It will only work if your current time beeing correct, because apt-get needs correct time.
So set the date temporaly with sudo date and then do
sudo apt-get install ntp

After this is done, proceed with the given command lines I gave you :)

**Comment by Phil1988** [comment]
@Cathobo  It seem like you are not sure what to do and what you acually do :D 
`sudo help` is not a valid command.
sudo just gives super user access and "help" is not a command.

`sudo chmod` does "chmod" (change directory permissions) with super user access.
chmod does have a few parameters, one of it is -x (see here: https://www.pluralsight.com/blog/it-ops/linux-file-permissions).

You are still not giving information where your error happens and this is very frustrating for anyonw willing to help (like me).

You should use the commands I provided. On one of the two `sudo chmod`  you get an "no such file or directory"-error, but its still open on which one.
Is it on "usr/sbin/chronyd" or on "/usr/sbin/ntpd".

Please provide more information also with screenshots etc. otherwise noone can help you and has to guess riddles.

**Comment by Cathobo** [comment]
Hi PhilThanks for your help, I have attached screenshots of the problem.. Screenshot 1: File gcode, Screenshot 2: QIDI Date settings , Screenshot 3: Uploaded g code file (wrong date), Screenshots 4 - 7  Putty terminal. Hope this helps.Neil
    On Wednesday, 3 January 2024 at 11:03:09 GMT, Phil1988 ***@***.***> wrote:  
 
 


@Cathobo It seem like you are not sure what to do and what you acually do :D
sudo help is not a valid command.
sudo just gives super user access and "help" is not a command.

sudo chmod does "chmod" (change directory permissions) with super user access.
chmod does have a few parameters, one of it is -x (see here: https://www.pluralsight.com/blog/it-ops/linux-file-permissions).

You are still not giving information where your error happens and this is very frustrating for anyonw willing to help (like me).

You should use the commands I provided. On one of the two sudo chmod you get an "no such file or directory"-error, but its still open on which one.
Is it on "usr/sbin/chronyd" or on "/usr/sbin/ntpd".

Please provide more information also with screenshots etc. otherwise noone can help you and has to guess riddles.

**Comment by QIDITECH** [comment]
Thank you for your feedback, a small number of motherboards do have this problem, we are still investigating this problem.

**Comment by Phil1988** [comment]
>  I have attached screenshots of the problem.. >
There is no screenshot attached here.
Please write directly on github, not as response via email.

look how your comments look like :)

**Comment by Cathobo** [comment]
Hi Phil here are the screen shots I sent
![Screenshot (1)](https://github.com/QIDITECH/QIDISlicer/assets/155291687/617f0747-ff52-414d-9770-646016e49152)
![Screenshot (2)](https://github.com/QIDITECH/QIDISlicer/assets/155291687/e72759d8-294c-487e-bec2-b32918661df4)
![Screenshot (3)](https://github.com/QIDITECH/QIDISlicer/assets/155291687/8ca8defa-3e9c-47ee-b3da-0d7eabafe9aa)
![Screenshot (4)](https://github.com/QIDITECH/QIDISlicer/assets/155291687/f5a3c5c8-fd7a-43ab-b4be-586ca2a33a21)
![Screenshot (5)](https://github.com/QIDITECH/QIDISlicer/assets/155291687/972ce660-f620-4ae5-8dc1-fbb20a0ab371)
![Screenshot (6)](https://github.com/QIDITECH/QIDISlicer/assets/155291687/4edb1373-d0d5-4ca8-ab48-934eadbb9383)
![Screenshot (7)](https://github.com/QIDITECH/QIDISlicer/assets/155291687/462ea8f2-b8a3-49b9-935c-50550cdcb2c1)

**Comment by Phil1988** [comment]
Ahh ok, then you need to
`sudo apt-get install chrony`
first.

retry again please and tell us if it worked or what was the error :)

**Comment by Cathobo** [comment]
Hi Phil,I installed chrony as you suggested and then entered the first line ok but now says cannot locate ntpd .
Neil
    On Saturday, 6 January 2024 at 12:44:28 GMT, Phil1988 ***@***.***> wrote:  
 
 


Ahh ok, then you need to
sudo apt-get install chrony
first.

retry again please and tell us if it worked or what was the error :)

**Comment by Phil1988** [comment]
Thats strange, because you said npt is already installed on your system...

you surely have done 
`sudo apt-get install ntp` ?

**Comment by Cathobo** [comment]
![Screenshot (7)](https://github.com/QIDITECH/QIDISlicer/assets/155291687/8ba20c0a-11d6-426b-860e-f0e796d78794)
Hi Phil, This is what happened after installing chron, it deleted ntp and says to install ntp, chron would be deleted.

**Comment by Phil1988** [comment]
And you did
`sudo apt-get install chrony` ?

**Comment by Cathobo** [comment]
![Screenshot (9)](https://github.com/QIDITECH/QIDISlicer/assets/155291687/2711af95-8cef-4262-b4f2-e2db0d926ff7)
Hi Phil Yes I did and it is still installed but npt is not. To install npt it would remove chron.
Neil

**Comment by Phil1988** [comment]
contact qidi directly about that...
Om running out of ideas

**Comment by Cathobo** [comment]
Thanks for your help Phil. An idea I have had is to run sudo date before I open the QIDI slicer, then the date would be correct. 
Regards Neil
    On Sunday, 7 January 2024 at 17:39:57 GMT, Phil1988 ***@***.***> wrote:  
 
 


contact qidi directly about that...
Om running out of ideas

**Comment by Cathobo** [comment]
Hi PhilGreat news, I have just booted my QIDI and without using the terminal to set date and time, I uploaded a file and Voila! the date and time is correct. I have attached a screenshot. It must be using the chron app I installed yesterday without NTP.
Regards Neil.
    On Monday, 8 January 2024 at 05:55:55 GMT, Neil Hobbs ***@***.***> wrote:  
 
  Thanks for your help Phil. An idea I have had is to run sudo date before I open the QIDI slicer, then the date would be correct. 
Regards Neil
    On Sunday, 7 January 2024 at 17:39:57 GMT, Phil1988 ***@***.***> wrote:  
 
 


contact qidi directly about that...
Om running out of ideas
